pkg/aggregator/time_sampler.go

```diff
@@ -29,38 +29,17 @@ type TimeSamplerID int
 // TimeSampler aggregates metrics by buckets of 'interval' seconds
 type TimeSampler struct {
 	interval                    int64
-	flushInterval               time.Duration
 	contextResolver             *timestampContextResolver
 	metricsByTimestamp          map[int64]metrics.ContextMetrics
 	counterLastSampledByContext map[ckey.ContextKey]float64
 	lastCutOffTime              int64
 	sketchMap                   sketchMap
 
-	// pointer to the shared MetricSamplePool stored in the Demultiplexer.
-	metricSamplePool *metrics.MetricSamplePool
-
 	// id is a number to differentiate multiple time samplers
 	// since we start running more than one with the demultiplexer introduction
-	id         TimeSamplerID
-	serializer serializer.MetricSerializer
-	stopChan   chan struct{}
-
-	// samples channel used to communicate from the calling routine to the one
-	// actively processing the samples
-	samples chan []metrics.MetricSample
-
-	// use this chan to command a flush of the time sampler
-	FlushChan chan FlushCommand
+	id TimeSamplerID
 
-	// parallel serialization configuration
-	parallelSerialization flushAndSerializeInParallel
-}
-
-// FlushCommand must be use to execute a flush of the TimeSampler.
-// If `BlockChan` is not nil, a message is sent when the flush is complete.
-type FlushCommand struct {
-	Time      time.Time
-	BlockChan chan struct{}
+	worker *timeSamplerWorker
```

Comment:
Would it make sense to remove this circular dependency (sampler and worker have pointers to each other)? If I understand correctly, Demultiplexer only interacts with the worker (via channels), so perhaps it would make sense to store worker pointers in the demultiplexer, leaving the sampler only as a field of the worker.